gaia/processes.py

```diff
@@ -0,0 +1,314 @@
+
+    def __init__(self, **kwargs):
+        """
+        Create a process to subset a raster by a vector polygon
+        :param clip_io: IO object containing vector polygon data
+        :param raster_io: IO object containing raster data
+        :param kwargs:
+        :return: SubsetProcess object
+        """
+        super(SubsetProcess, self).__init__(**kwargs)
+        if not self.output:
+            self.output = RasterFileIO(name='result',
+                                       uri=self.get_outpath())
+
+    def compute(self):
+        super(SubsetProcess, self).compute()
+        clip_df = self.inputs[0].read()
+        raster_img = self.inputs[1].read()
+        # Merge all features in vector input
+        raster_output = self.output.uri
+        self.output.create_output_dir(raster_output)
+        clip_json = clip_df.geometry.unary_union.__geo_interface__
+        self.output.data = gdal_clip(raster_img, raster_output, clip_json)
+
+
+class WithinProcess(GaiaProcess):
+    """
+    Similar to SubsetProcess but for vectors: calculates the features within
+    a vector dataset that are within (or whose centroids are within) the
+    polygons of a second vector dataset.
+    """
+
+    required_inputs = (('first', formats.VECTOR), ('second', formats.VECTOR))
+    default_output = formats.JSON
+
+    def __init__(self, **kwargs):
+        super(WithinProcess, self).__init__(**kwargs)
+        if not self.output:
+            self.output = VectorFileIO(name='result',
+                                       uri=self.get_outpath())
+
+    def compute(self):
+        super(WithinProcess, self).compute()
+        # TODO: Don't assume GeoPandas Dataframe. Could be PostGIS,Girder,etc.
+        if len(self.inputs) != 2:
+            raise GaiaException('WithinProcess requires 2 inputs')
+        first_df = self.inputs[0].read()
+        second_df = self.inputs[1].read()
+        first_within = first_df[first_df.geometry.within(
+            second_df.geometry.unary_union)]
+        self.output.data = first_within
+        self.output.write()
+
+
+class IntersectsProcess(GaiaProcess):
+    """
+    Calculates the features within the first vector dataset that touch
+    the features of the second vector dataset.
+    """
+
+    default_output = formats.JSON
+
+    def __init__(self, **kwargs):
+        super(IntersectsProcess, self).__init__(**kwargs)
+        if not self.output:
+            self.output = VectorFileIO(name='result',
+                                       uri=self.get_outpath())
+
+    def compute(self):
+        super(IntersectsProcess, self).compute()
+        first_df = self.inputs[0].read()
+        second_df = self.inputs[1].read()
+        first_intersects = first_df[first_df.geometry.intersects(
+            second_df.geometry.unary_union)]
+        self.output.data = first_intersects
+        self.output.write()
+        logger.debug(self.output)
+
+
+class DifferenceProcess(GaiaProcess):
```

Comment:
@mbertrand I am wondering if these will go into their own separate file in the future specifially because these processes will have dependencies.